### DISCUSSION TITLE
Syndicated Photos library

### DISCUSSION THREAD

**Discussion by RhetTbull** [RhetTbull/osxphotos#1719] [General]
In doing some debugging for another issue today I discovered a "hidden" Photos library at `~/Library/Photos/Libraries/Syndication.photoslibrary`. This appears to include all the photos that are "syndicated" which appears to mean "shared with you via Messages or another app".  OSXPhotos already reads syndicated photos in the main library but I was unaware there was an entire separate library maintained by the system. OSXPhotos can of course read this library. Haven't had time to poke around too much but this might be of interest to some OSXPhotos users.  There are also privacy implications with this as any photo shared in a message appears to be in here.

**Comment by edhutchison** [comment]
Just started exploring the Syndication Photo Library.  Have you found anything new since October 2024?  Have you run osxphotos on it?  I did, to compare it to my system photo library and the Syndication Photos Library contained significantly more photos than I could see reported anywhere that made sense in the System Photos Library's info output.  Do you know what the discrepancy might be?  Also, do you know if osxphotos can be used to download / copy all of the Syndicated Photos into the System Photos Library?

macOS 15.7.7
Photos 10.0
osxphotos, version 0.76.1
Python 3.14.6 (main, Jun 10 2026, 10:03:53) [Clang 17.0.0 (clang-1700.6.4.2)]
Python executable: /Users/edhutchison/.local/share/uv/tools/osxphotos/bin/python
macOS 15.7.7, x86_64

DB Version: 5001, DB Model Version: 18600, Photos version: 10


excerpt of diff between Syndicated and System Photo libraries:

```
Total Assets (includes hidden, shared, syndicated which do no   Total Assets (includes hidden, shared, syndicated which do no
total: 18248, photos: 17341, videos: 907                      | total: 8390, photos: 7935, videos: 455
Photo App Totals (excludes hidden, shared, syndicated)          Photo App Totals (excludes hidden, shared, syndicated)
total: 15902, photos: 15102, videos: 800                      | total: 0, photos: 0, videos: 0
In Trash (Recently Deleted album)                               In Trash (Recently Deleted album)
total: 0                                                        total: 0
Hidden (assets in hidden albums)                                Hidden (assets in hidden albums)
total: 1, photos: 1, videos: 0                                | total: 0, photos: 0, videos: 0
Missing (assets which are missing from the library, for examp   Missing (assets which are missing from the library, for examp
total: 0, photos: 1056, videos: 72                            | total: 0, photos: 7931, videos: 455
Cloud Assets (includes assets tracked by iCloud but not share   Cloud Assets (includes assets tracked by iCloud but not share
total: 15200, in iCloud: 14943, not uploaded to iCloud: 257,  | total: 0, in iCloud: 0, not uploaded to iCloud: 0, not downlo
Referenced Files (files which have not been copied to the Pho   Referenced Files (files which have not been copied to the Pho
total: 960, photos: 960, videos: 0                            | total: 0, photos: 0, videos: 0
Shared Library (photos shared via iCloud shared library)        Shared Library (photos shared via iCloud shared library)
total: 0, photos: 0, videos: 0                                  total: 0, photos: 0, videos: 0
Shared (photos shared via iCloud shared albums)                 Shared (photos shared via iCloud shared albums)
total: 1216, photos: 1181, videos: 35                         | total: 0, photos: 0, videos: 0
Shared Moments (moments shared in Messages)                     Shared Moments (moments shared in Messages)
total: 0                                                        total: 0
Syndicated (photos shared via Messages or other apps)           Syndicated (photos shared via Messages or other apps)
total: 1211, saved to library: 82, not saved to library: 1129 | total: 8390, saved to library: 0, not saved to library: 8390
Favorite (photos marked as favorites)                           Favorite (photos marked as favorites)
total: 1714, photos: 1658, videos: 56                         | total: 0, photos: 0, videos: 0
Edited                                                          Edited
total: 1181, photos: 949, videos: 232                         | total: 0, photos: 0, videos: 0
edited in external application: 1                             | edited in external application: 0
```

**Reply by edhutchison** [reply]
Left column is System Photo Library and Right column is Syndication Photo Library.

**Reply by RhetTbull** [reply]
I've not had a chance to dig into this.
